sections/4.1-error-handling.tex

```diff
@@ -94,15 +94,27 @@ \subsection{Promise 1: No null pointer dereferences}
     Ok(a + b)
 }
 \end{minted}
-\pause
+Note: Error signature must match!
+\end{frame}
+
+%%%
+
+\begin{frame}[fragile]{Ergonomic error handling with the \texttt{?} operator}
+\begin{minted}{rust}
+fn do_calc() -> Result<i32, String> {
+    let a = do_subcalc1()?;
+    let b = do_subcalc2()?;
```

Comment:
I think this is a rather bad example for the `?` operator. It's more useful in my opinion when chaining is involved:
```Rust
// ugliness of nested tries
let a = try!(try!(do_foo()).do_bar());
// This is where ? shines
let a = do_foo()?.do_bar()?;
```